.github/workflows/check-by-name.yml

```diff
@@ -57,117 +32,47 @@ jobs:
           fetch-depth: 2
       - name: Determining PR git hashes
         run: |
-          # For pull_request_target this is the same as $GITHUB_SHA
-          echo "baseSha=$(git rev-parse HEAD^1)" >> "$GITHUB_ENV"
-
-          echo "headSha=$(git rev-parse HEAD^2)" >> "$GITHUB_ENV"
+          # Please don't put any logic here.  To ensure local
+          # reproducibility, this file should only print diagnostics
+          # (`env`, `set -x`) and execute scripts found elsewhere in
+          # nixpkgs.
+          env
+          set -x
+          exec pkgs/test/nixpkgs-check-by-name/workflows/check-by-name/determine-pr-hashes.sh
       - uses: cachix/install-nix-action@v23
       - name: Determining channel to use for dependencies
         run: |
-          echo "Determining which channel to use for PR base branch $GITHUB_BASE_REF"
-          if [[ "$GITHUB_BASE_REF" =~ ^(release|staging|staging-next)-([0-9][0-9]\.[0-9][0-9])$ ]]; then
-              # Use the release channel for all PRs to release-XX.YY, staging-XX.YY and staging-next-XX.YY
-              channel=nixos-${BASH_REMATCH[2]}
-              echo "PR is for a release branch, using release channel $channel"
-          else
-              # Use the nixos-unstable channel for all other PRs
-              channel=nixos-unstable
-              echo "PR is for a non-release branch, using unstable channel $channel"
-          fi
-          echo "channel=$channel" >> "$GITHUB_ENV"
+          # Please don't put any logic here.  To ensure local
+          # reproducibility, this file should only print diagnostics
+          # (`env`, `set -x`) and execute scripts found elsewhere in
+          # nixpkgs.
+          env
+          set -x
+          exec pkgs/test/nixpkgs-check-by-name/workflows/check-by-name/determine-channel-for-dependencies.sh
```

Comment:
It's not great how there's all of these separate files. The main reason these steps are split up so that it's easier to see where time is spent (e.g. see https://github.com/NixOS/nixpkgs/actions/runs/6844035438/job/18607346212). But that's not very important, I'd rather have this and all steps after as a single script.

Also we shouldn't make `env` necessary. If we have a script to reproduce it locally, the script shouldn't depend on GitHub-specific variables.

I also don't think `set -x` is necessary, the code is very verbose and already prints everything useful.